docs/concepts/pull-dependencies.md

```diff
@@ -0,0 +1,202 @@
+      "dfx_init": null,
+      "candid_args": "()"
+    },
+    "yahli-baaaa-aaaaa-aabtq-cai": {
+      "name": "dep_c",
+      "deps": [
+        "yofga-2qaaa-aaaaa-aabsq-cai"
+      ],
+      "wasm_url": "example.com/c.wasm",
+      "wasm_hash": "016df9800dc5760785646373bcb6e6bb530fc17f844600991a098ef4d486cf0b",
+      "dfx_init": "Nat",
+      "candid_args": "(nat)"
+    }
+  }
+}
+```
+
+There are three dependencies:
+
+- "yhgn4-myaaa-aaaaa-aabta-cai": "dep_b" in `dfx.json`;
+- "yahli-baaaa-aaaaa-aabtq-cai": "dep_c" in `dfx.json`;
+- "yofga-2qaaa-aaaaa-aabsq-cai": an indirect dependency that both "dep_b" and "dep_c" depend on;
+
+### 3. Set init arguments using `dfx deps init`
+
+Running `dfx deps init` will iterate all dependencies in `pulled.json`, set empty argument for the ones need no init argument and print the list of dependencies that require init argument.
+
+Then running `dfx deps init <CANISTER> --argument <ARGUMENT>` to set init argument for individual dependency.
+
+The init arguments will be recorded in `deps/init.json`.
+
+For our example, we should run:
+
+```
+> dfx deps init
+WARN: The following canister(s) require an init argument. Please run `dfx deps init <NAME/PRINCIPAL>` to set them individually:
+yofga-2qaaa-aaaaa-aabsq-cai
+yahli-baaaa-aaaaa-aabtq-cai (dep_c)
+> dfx deps init yofga-2qaaa-aaaaa-aabsq-cai --argument 10
+> dfx deps init deps_c --argument 20
+```
+
+The generated `init.json` has following content:
+
+```json
+{
+  "canisters": {
+    "yofga-2qaaa-aaaaa-aabsq-cai": {
+      "arg_str": "10",
+      "arg_raw": "4449444c00017d0a"
+    },
+    "yhgn4-myaaa-aaaaa-aabta-cai": {
+      "arg_str": null,
+      "arg_raw": null
+    },
+    "yahli-baaaa-aaaaa-aabtq-cai": {
+      "arg_str": "20",
+      "arg_raw": "4449444c00017d14"
+    }
+  }
+}
+```
+
+### 4. Deploy pull dependencies on local replica using `dfx deps deploy`
+
+Running `dfx deps deploy` will:
+
+1. create the dependencies on the local replica with the same mainnet canister ID;
+2. install the downloaded wasm with the init arguments in `init.json`;
+
+You can also specify the name or principal to deploy one particular dependency.
+
+For our example:
+```
+> dfx deps deploy
+Creating canister: yofga-2qaaa-aaaaa-aabsq-cai
+Installing canister: yofga-2qaaa-aaaaa-aabsq-cai
+Creating canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+Installing canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+Creating canister: yahli-baaaa-aaaaa-aabtq-cai (dep_c)
+Installing canister: yahli-baaaa-aaaaa-aabtq-cai (dep_c)
+> dfx deps deploy yofga-2qaaa-aaaaa-aabsq-cai
+Creating canister: yofga-2qaaa-aaaaa-aabsq-cai
+Installing canister: yofga-2qaaa-aaaaa-aabsq-cai
+> dfx deps deploy dep_b
+Creating canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+Installing canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+```
+
+**Note**
+
+- `dfx deps deploy` always create the canister with the anonymous identity so that dependencies and application canisters will have different controllers;
```

Comment:
```suggestion
- `dfx deps deploy` always creates the canister with the anonymous identity so that dependencies and application canisters will have different controllers;
```